examples/systemcalls/systemcalls.c

```diff
@@ -80,20 +99,53 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
         command[i] = va_arg(args, char *);
     }
     command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
 
+    int status;
+    pid_t pid;
+    int fd;
 
-/*
- * TODO
- *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
- *   redirect standard out to a file specified by outputfile.
- *   The rest of the behaviour is same as do_exec()
- *
-*/
+    fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+    if(fd < 0)
+    {
+        perror("open");       
+        return false;
+    }
+
+    pid = fork();
 
-    va_end(args);
+    if(pid == -1)
+    {
+        perror("fork");
+        return false;
+    }
+    else if(pid == 0)
+    {
+        if(dup2(fd,1) < 0)
+        {
+            perror("dup2");
+            printf("\nCannot duplicate file descriptor\n");
+            
+            return false;
```

Comment:
@kevintom98 return may not do what you want here since you are in the child process. You need to exit with error so you could catch the error in the parent process.